src/test/java/io/vertx/core/parsetools/JsonParserTest.java

```diff
@@ -780,7 +781,17 @@ public void testStreamFetchInHandler() {
 
   @Test
   public void testStreamEnd() {
-    FakeStream stream = new FakeStream();
+    FakeStream stream = new FakeStream(){
+      //regression check for #2790 - ensure that by accident resume method is not called.
+      @Override
```

Comment:
rather add on FakeStream a pauseCount / resumeCount that are incremented when those methods are called and check the counters are == 0 at the end of the test

Reply:
But that is done in FakeReadStream - should I use it over FakeStream ?

Reply:
done